### ISSUE TITLE
`particle flash` add `--system` flag

### ISSUE THREAD

**Issue by kennethlimcp** [spark/particle-cli#32]
It gets tedious trying to flash a new system firmware to a Photon since the current CLI does not support it.

- `particle flash --system xxx.bin`
- system address: `0x8020000` + `-a 0`

**Comment by berkley** [comment]
+1.  Would love to see a fix for this.  The CLI is extremely useful for my projects and I can't use it with my photons because of this issue.

**Comment by kennethlimcp** [comment]
@berkley you don't need this flag unless you are doing a system upgrade. For normal user firmware upgrade it's `particle flash --usb xxxx.bin`

**Comment by berkley** [comment]
Ahh, right.  I misunderstood this particular issue.  I'm really interested in being able to compile for the photon via the CLI, which the README still says will be fixed soon.  Maybe I should open an issue for that.

**Comment by brycekahle** [comment]
@kennethlimcp You would need flags for both system part 1 and system part 2 right?

**Comment by kennethlimcp** [comment]
I guess that is done using `particle update` already?

**Comment by brycekahle** [comment]
Yes, but that command is hard-coded to a specific version.

**Comment by kennethlimcp** [comment]
Yes right. Something like`particle flash --system1` and `particle system2` might be good thought syntax sounds a little weird.

**Comment by m-mcgowan** [comment]
Hey Guys! The CLI can examine the binary and know if it's part1 or part2 using the binary-image-reader lib. so ideally you can just write

```
particle flash --usb system-part1.bin
particle flash --usb system-part2.bin
particle flash --usb tinker.bin
```
and all 3 binaries end up in the right place.  Since there's no special flags to denote type of image/location, less room for user error.

**Comment by kennethlimcp** [comment]
Won't mind helping to build though the existence of that library is not known to me until now.

**Comment by chobberoni** [comment]
@kennethlimcp you're great man! truly great.

**Comment by towynlin** [comment]
:+1:

**Comment by brycekahle** [comment]
Fixed as part of #159. This is automatic now.
